### ISSUE TITLE
Implement CLA check bypass based on the license attached to a commit

### ISSUE THREAD

**Issue by simondeziel** [canonical/has-signed-canonical-cla#34]
tl;dr: commits with `License: Apache-2.0` should be considered as if the CLA was signed.

Long version:
With the LXD project moving to Canonical (https://discourse.ubuntu.com/t/faq-lxd-has-been-moved-to-canonical/36709) and changing its license (Apache-2.0 -> AGPL-3.0-only), a CLA check was added.

IANAL, but the license change was apparently possible because the old Apache-2.0 license implicitly granted the same rights to Canonical as signing the CLA would. Thus, Apache-2.0 contributions/commits should be exempted from requiring the author to sign the CLA.

In the LXD project, we would like to be able to cherry-pick commits contributed to any Apache-2.0 projects and incorporate them into LXD. When doing those cherry-picks, we add a `License: Apache-2.0` trailer to the commit to convey they were contributed under that license all the while retaining the original author/signed-off-by. That original author/signed-off-by sometimes has not sign the CLA meaning its commit would not pass the `canonical/has-signed-canonical-cla` check.

One such example is https://github.com/canonical/lxd/pull/12763 where a Canonical employee cherry-picked a commit contributed to Incus (project under Apache-2.0 license) by a community member that has not signed the CLA. The CLA action thus failed like that:

```
Run canonical/has-signed-canonical-cla@v1
Installing python3-launchpadlib

Checking the following users on GitHub:
- jalbstmeijer ✕ (issue checking CLA status [HttpError: User does not exist or is not a member of the organization])
- ru-fu ✓ (@canonical.com account)

Checking the following users on Launchpad:
- justin@webslice.nl ✕ (has no Launchpad account)

Error: CLA Check - FAILED
```

The cherry-picked commit (https://github.com/canonical/lxd/pull/12763/commits/e54e9fd6fdb6c1c283b1802b469a10edc1f76c9d) has the following commit message/metadata:

```
$ git show --no-patch e54e9fd6fdb6c1c283b1802b469a10edc1f76c9d
commit e54e9fd6fdb6c1c283b1802b469a10edc1f76c9d
Author: jalbstmeijer <justin@webslice.nl>
Date:   Mon Jan 22 16:28:32 2024 +0100

    Add note about scrape_interval and update examples
    
    Signed-off-by: jalbstmeijer <justin@webslice.nl>
    (cherry picked from commit 4134c2925f9982d90663b7b6656bb5110058f16e)
    Signed-off-by: Ruth Fuchss <ruth.fuchss@canonical.com>
    License: Apache-2.0
```

This ticket is thus asking for an additional action parameter (`implicit-approval-from-licenses` maybe?) that would accept a list of licenses identifiers that are deemed to provide implicitly approval of the CLA requirements. It could potentially default to `Apache-2.0`.

**Comment by simondeziel** [comment]
Fixed by #35
